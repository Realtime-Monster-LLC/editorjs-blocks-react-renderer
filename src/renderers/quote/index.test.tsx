import React from 'react';
import { create } from 'react-test-renderer';
import Quote, { QuoteBlockData } from '.';

describe('<Quote />', () => {
  describe('when receives a Quote block', () => {
    const data: QuoteBlockData = {
      text: '<i>Esse ea consectetur est fugiat ut dolor pariatur ex voluptate ad Lorem Lorem sit sunt.</i>\n\n<i>Eiusmod voluptate <b>ullamco</b> laborum minim elit.</i><i><br></i>Cillum ullamco fugiat tempor dolore enim.\nExcepteur exercitation amet aliqua proident labore qui sint do dolore sint nulla aute.</i>',
      caption: 'Carpe Diem, <cite>Lorem ipusm dolor</cite>.',
      alignment: 'center',
    };

    it('renders a <blockquote> block with text and <footer>', () => {
      expect(create(<Quote data={data} className="special-quote" />).toJSON()).toMatchSnapshot();
    });
  });
});
