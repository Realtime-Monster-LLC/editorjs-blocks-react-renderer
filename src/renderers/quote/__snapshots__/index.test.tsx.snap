// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> when receives a Quote block renders a <blockquote> block with text and <footer> 1`] = `
<blockquote
  className="text-align-center special-quote"
>
  <p>
    Esse ea consectetur est fugiat ut dolor pariatur ex voluptate ad Lorem Lorem sit sunt.
  </p>
  <p>
    Eiusmod voluptate ullamco laborum minim elit.
  </p>
  <p>
    Cillum ullamco fugiat tempor dolore enim.
    <br />
    Excepteur exercitation amet aliqua proident labore qui sint do dolore sint nulla aute.
  </p>
  <footer>
    Carpe Diem, 
    <cite>
      Lorem ipusm dolor
    </cite>
    .
  </footer>
</blockquote>
`;
